pkg/controller/sparkapplication/controller.go

```diff
@@ -523,17 +567,149 @@ func (c *Controller) updateSparkApplicationStatusWithRetries(
 	}
 
 	if lastUpdateErr != nil {
-		glog.Errorf("failed to update SparkApplication %s: %v", toUpdate.Name, lastUpdateErr)
+		glog.Errorf("failed to update SparkApplication %s: %v", original.Name, lastUpdateErr)
 	}
 
 	return nil
 }
 
+// isValidDriverStatusTransition determines if the driver state transition is valid
+func isValidDriverStatusTransition(oldStatus, newStatus v1alpha1.ApplicationStateType) bool {
+	for _, validStatus := range validDriverStateTransitions[oldStatus] {
+		if newStatus == validStatus {
+			return true
+		}
+	}
+	return false
+}
+
+// isValidExecutorStatusTransition determines if the executor state transition is valid
+func isValidExecutorStatusTransition(oldStatus, newStatus v1alpha1.ExecutorState) bool {
+
```

Comment:
Please remove the extra empty line here and in several places below.

Reply:
done